### PR TITLE
`opentelemetry-instrumentation-system-metrics` restore package

### LINKED ISSUE #1011: Restore instrumentation/system-metrics
The `opentelemetry-instrumentation-system-metrics` package exists against the old metrics API/SDK. This issue is to restore the package and update it to use the new API/SDK.

### PR DESCRIPTION
**Pull request by codeboten**
# Description

Restore the system-metrics instrumentation package into main from the `metrics` branch and update its usage of the api/sdk. Leaving in draft until the tests are fixed

Fix #1011

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+from opentelemetry.instrumentation.system_metrics.package import _instruments
+from opentelemetry.instrumentation.system_metrics.version import __version__
+from opentelemetry.sdk.util import get_dict_as_key
+
+
+class SystemMetricsInstrumentor(BaseInstrumentor):
+    # pylint: disable=too-many-statements
+    def __init__(
+        self,
+        labels: Optional[Dict[str, str]] = None,
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:91]: It looks like this just adds extra attributes to all measurements. We should probably remove this?

We can do it in a separate PR though as this is just restoring the package

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+        self._labels = {} if labels is None else labels
+        self._meter = None
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs):
+        meter_provider = kwargs.get("meter_provider")
+        self._meter = get_meter(
+            "io.otel.python.instrumentation.system_metrics",
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:129]: is this consistent with other insturmentation packages?

File: instrumentation/opentelemetry-instrumentation-system-metrics/tests/test_system_metrics.py [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+
+        system_metrics = SystemMetricsInstrumentor()
+        system_metrics.instrument(meter_provider=meter_provider)
+        self._assert_metrics(observer_name, reader, expected)
+        system_metrics.uninstrument()
+
+    # When this test case is executed, _get_system_cpu_utilization gets run
+    # too because of the controller thread which runs all observers. This patch
+    # is added here to stop a warning that would otherwise be raised.
+    # pylint: disable=unused-argument
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-system-metrics/tests/test_system_metrics.py:128]: Is htis comment still accurate?

**Review by aabmass — APPROVED** [review]
I think the configuration mechanism in this instrumentation is largely superseded by views. Wdyt @codeboten?

Approving since most of this code is being restored. Thanks for undertaking this Alex!

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 2 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py]: i've reverted back to using `__name__` as per other instrumentations in this repo... we should consider if this is correct moving forward though.

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 1 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:91]: Agree, happy to do this as a separate item

File: instrumentation/opentelemetry-instrumentation-system-metrics/tests/test_system_metrics.py (hunk 3 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/tests/test_system_metrics.py]: The patch is still needed, but i've updated the comment.

File: instrumentation/opentelemetry-instrumentation-system-metrics/setup.cfg [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    Programming Language :: Python
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+
+[options]
+python_requires = >=3.5
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-system-metrics/setup.cfg:38]:
```suggestion
python_requires = >=3.6
```

File: instrumentation/opentelemetry-instrumentation-system-metrics/setup.cfg [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+package_dir=
+    =src
+packages=find_namespace:
+install_requires =
+    opentelemetry-api ~= 1.10.0
+    opentelemetry-sdk ~= 1.10.0
+    psutil ~= 5.7.0
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-system-metrics/setup.cfg:45]: Curious why ~= 5.7.0 here? README and _instruments has >= 5.7.3? Also does it not work with other minor versions like 5.6?

**Review by codeboten — COMMENTED** [review]
Added an issue to track the work needed to support pypy3: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1033

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 1 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:91]: Added an issue: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1034

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 2 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py]: Issue added: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1035

File: instrumentation/opentelemetry-instrumentation-system-metrics/setup.cfg (hunk 5 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/setup.cfg]: I've tested with psutil 5.0.0 and everything seems to work, i've updated the dependencies. PTAL @srikanthccv

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,23 +7,29 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased](https://github.com/open-telemetry/opentelemetry-python/compare/v1.10.0-0.29b0...HEAD)
 
+### Fixed
+
+- `opentelemetry-instrumentation-flask` Fix non-recording span bug
+  ([#999])(https://github.com/open-telemetry/opentelemetry-python-contrib/pull/999)
+- `opentelemetry-instrumentation-tornado` Fix non-recording span bug
+  ([#999])(https://github.com/open-telemetry/opentelemetry-python-contrib/pull/999)
+- `opentelemetry-instrumentation-pyramid` Handle non-HTTPException exceptions
```
Comment (lzchen) [inline CHANGELOG.md:16]: Is this not an additive feature?

**Review by srikanthccv — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+class SystemMetricsInstrumentor(BaseInstrumentor):
+    # pylint: disable=too-many-statements
+    def __init__(
+        self,
+        labels: Optional[Dict[str, str]] = None,
+        config: Optional[Dict[str, List[str]]] = None,
+    ):
+        super().__init__()
+        if config is None:
+            self._config = {
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:96]: Nit: Could this be defined outside of the constructor as a constant default?

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 8]
```diff
@@ -0,0 +1,647 @@
...
+        self._meter.create_observable_gauge(
+            callback=self._get_system_swap_utilization,
+            name="system.swap.utilization",
+            description="System swap utilization",
+            unit="1",
+        )
+
+        # # self._meter.create_observable_counter(
+        # #     callback=self._get_system_swap_page_faults,
+        # #     name="system.swap.page_faults",
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:210]: Are these TODOs?

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 9]
```diff
@@ -0,0 +1,647 @@
...
+
+        self._meter.create_observable_counter(
+            callback=self._get_runtime_gc_count,
+            name=f"runtime.{self._python_implementation}.gc_count",
+            description=f"Runtime {self._python_implementation} GC count",
+            unit="bytes",
+        )
+
+    def _uninstrument(self, **__):
+        pass
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:317]: I would assume `uninstrument()` would cause the automatic collection to stop? Perhaps add a TODO?

Comment (srikanthccv) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:317]: relevant spec issue https://github.com/open-telemetry/opentelemetry-specification/issues/2232

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 10]
```diff
@@ -0,0 +1,647 @@
...
+
+    def _get_system_cpu_time(self) -> Iterable[Measurement]:
+        """Observer callback for system CPU time
+
+        Args:
+            observer: the observer to update
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:323]: Nit: Do callbacks have to have `args` docstrings?

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 7 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:96]: I'll do this w/ the pr to remove labels arg

**Review by lzchen — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 10 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:323]: Good catch, removed.

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 8 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py]: Grouped TODOs in the instrument method. I don't know if we need issues for each one or if we wait until someone asks for the feature?

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py (hunk 9 above)
Comment (codeboten) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:319]: Added an issue: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1036
